OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -183,7 +186,26 @@ Order IIssueDeployOrder.IssueDeployOrder(Actor self, bool queued)
 			return new Order("GrantConditionOnDeploy", self, queued);
 		}
 
-		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued) { return !IsTraitPaused && !IsTraitDisabled; }
+		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued)
+		{
+			if (IsTraitPaused || IsTraitDisabled || self.IsDead || self.Disposed)
+				return false;
+
+			if (queued || !Info.SmartDeploy || DeployState == DeployState.Undeployed || DeployState == DeployState.Undeploying)
+				return true;
+
+			foreach (var actor in self.World.Selection.Actors)
+			{
+				if (actor == self || actor.IsDead || !actor.IsInWorld)
+					continue;
+
+				if (actor.TraitsImplementing<GrantConditionOnDeploy>()
+					.Any(d => !d.IsTraitPaused && !d.IsTraitDisabled && (d.DeployState == DeployState.Undeployed || d.DeployState == DeployState.Undeploying)))
```

Comment:
Found another way SmartDeploy can lead to the button not doing anything, tho not sure how to fix. This one is related to terrain checks, in a scenario like below, nothing happens because it is trying to deploy the one on land, which can't deploy due to terrain.

![image](https://github.com/user-attachments/assets/897ebe8d-aa97-4e51-8819-0814e8d59584)

Also i'm not sure if it should tbh. In a case like Thumper it makes sense, but you may also end up undeploying all your Tick Tanks in the middle of combat, because one undeployed one that you tried to tell to deploy is on a ramp. Perhaps a better way to handle it could be adding ability to get the unit to find a valid terrain nearby to deploy on when given the order, but that's likely out of scope for this PR as well.

Reply:
this feels like its functioning as intended tbh